### DISCUSSION TITLE
Renovate cannot install Go 1.22.0

### DISCUSSION THREAD

**Discussion by uhthomas** [renovatebot/renovate#27513] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

v37.203.4

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Server

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

Similar to https://github.com/renovatebot/renovate/discussions/23763, but not the same.

PR fails because Renovate cannot install Go 1.22.0.

<img width="788" alt="Screenshot 2024-02-22 at 18 28 06" src="https://github.com/renovatebot/renovate/assets/9749173/7915372b-4bf6-470d-b677-2fbd2e7a55ed">

Raw output from the command:

```
err.stdout=[18:07:04.175] INFO (903): Installing tool golang@1.22.0...
tool golang v1.22.0 is already installed
linking tool golang v1.22.0
No /opt/containerbase/tools/golang/1.22.0/bin defined - aborting
```

I can't reproduce this with the hosted version for github.com.

### Relevant debug logs

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment]
It looks like there's a corrupted install. Can you reset the container?

**Reply by uhthomas** [reply]
Thanks for the suggestion @rarkins, we did this and it seemed to resolve it for a while. Has unfortunately happened again.

```sh
err.stdout=[06:20:40.391] INFO (10536): Installing tool golang@1.22.5...
tool golang v1.22.5 is already installed
linking tool golang v1.22.5
No /opt/containerbase/tools/golang/1.22.5/bin defined - aborting
[06:20:40.427] FATAL (10536): Command failed with exit code 1: /usr/local/containerbase/bin/install-tool.sh golang 1.22.5
```

Is there anything which can be done to make this more robust?

**Reply by rarkins** [reply]
@viceice have you been able to add the retry to containerbase yet?

**Reply by viceice** [reply]
yes, should already inside the hosted app for a while and on the oss images

**Reply by uhthomas** [reply]
Which version, sorry? We are currently on 37.407.1.

**Reply by uhthomas** [reply]
Is this what is being referenced?

- https://github.com/renovatebot/renovate/discussions/29999

**Reply by viceice** [reply]
it's fixed since renovate v37.427.0, hosted app should have it since 2024-07-14

- https://github.com/renovatebot/renovate/pull/30113#issuecomment-2219850275
- https://github.com/renovatebot/base-image/releases/tag/2.25.0

**Reply by viceice** [reply]
- https://github.com/containerbase/base/pull/2963
that should fix corrupt cache files
